libraries/plugins/transaction_status/transaction_status_plugin.cpp

```diff
@@ -25,11 +48,21 @@ class transaction_status_impl
    void on_post_apply_block( const block_notification& note );
 
    chain::database&              _db;
-   uint32_t                      block_depth;
-   uint32_t                      track_after_block;
+   uint32_t                      nominal_block_depth = 0;       //!< User provided block-depth
+   uint32_t                      nominal_track_after_block = 0; //!< User provided track-after-block
+   uint32_t                      actual_block_depth = 0;        //!< Calculated block-depth
+   uint32_t                      actual_track_after_block = 0;  //!< Calculated track-after-block
    bool                          tracking = false;
+   bool                          rebuild_state_flag = false;
    boost::signals2::connection   post_apply_transaction_connection;
    boost::signals2::connection   post_apply_block_connection;
+   bool                          state_is_valid();
```

Comment:
Do `state_is_valid` and `rebuild_state` need to be made public? What use case is there for them being accessible here? Are they needed on the `transaction_status_plugin` struct at all?

Reply:
If it is only for tests, surround it in `#ifdef IS_TEST_NET`

Public/private specifiers in C++ are somewhat meaningless/easy to refactor. Putting private methods on an impl that lives only is source is a much more effective way of enforcing privacy of members.

Reply:
Yes they are used just for testing. I wrapped the functions in `transaction_status_plugin` with `#ifdef IS_TEST_NET`, and left them  alone in the `transaction_status_impl` class.